### PR TITLE
Add AddMissingFunKeyword code fix

### PR DESCRIPTION
**Pull request by cartermp**
This one was `fun`. Get it? har har har har har har har har har har har har har

Anyways, this is what it looks like:

image

And:

image

The fact that the error that triggers this is a parse error makes it so that we have to use our nice friend, `Tokenizer.getSymbolAtPosition`. We may want to consider moving that into FCS proper, since it lets you resolve a "symbol" without needing to parse or typecheck something. Very nice when you want the range of that "symbol".

### CODE REVIEW

**Comment by alfonsogarciacaro** [comment]
Can/Will this work with fat arrows? I guess that would be the common mistake made by newcomers.

File: vsintegration/src/FSharp.Editor/CodeFix/AddMissingFunKeyword.fs [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+            let defines = CompilerEnvironment.GetCompilationDefinesForEditing parsingOptions
+
+            let adjustedPosition =
+                let rec loop str pos =
+                    if not (String.IsNullOrWhiteSpace(str)) then
+                        pos
+                    else
+                        loop (sourceText.GetSubText(TextSpan(pos - 1, 1)).ToString()) (pos - 1)
+
+                loop (sourceText.GetSubText(TextSpan(context.Span.Start - 1, 1)).ToString()) context.Span.Start
```
Comment (TIHan) [inline vsintegration/src/FSharp.Editor/CodeFix/AddMissingFunKeyword.fs:46]: Is it possible that the construction of TextSpan would receive any invalid input? Such as, `context.Span.Start - 1` , would ever be `-1`?

Comment (cartermp) [reply vsintegration/src/FSharp.Editor/CodeFix/AddMissingFunKeyword.fs:46]: Not at this point, no. You can of course write literally `->` as the first set of characters in a script file, but it won't trigger this error when that happens. The compiler thinks it's in the middle of a `yield` expression to be precise, which is weird, but unrelated to this or any other code fixer.

Comment (cartermp) [reply vsintegration/src/FSharp.Editor/CodeFix/AddMissingFunKeyword.fs:46]: Sorry - it _does_ trigger FS0010, but the span isn't actually `->`, so it never reaches here in the first place

**Review by KevinRansom — APPROVED** [review]

**Pull request merged**